File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.js
/**
 * @param {number[]} candies
 * @param {number} extraCandies
 * @return {boolean[]}
 */
var kidsWithCandies = function(candies, extraCandies) {
  const result = [];
  const max = Math.max(...candies);
  
  candies.forEach(el => {
    if (el + extraCandies >= max) {
      result.push(true)
    } else {
      result.push(false)
    }
  })
  
  return result;
};

File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.py
class Solution:
  def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
    maximum = max(candies)
    result = []

    for i in range(0, len(candies)):
      cur = candies[i]

      if cur + extraCandies >= maximum:
        result.append(True)
      else:
        result.append(False)

    return result
